### PR TITLE
INTYGFV-15077: Implementation of categories Hörsel och balanssinne -> Demens och andra kognitiva störningar

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
Still under (WIP)

Tho I feel that this feature is has been growing out of proporsion, therefore I create this PR to give people the chance to partially look throught it. I hope this will mitigate the workload of reviewing this PR in the future.,

### CODE REVIEW

File: ts/ts-bas/src/test/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionBeskrivningRiskfaktorerTest.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            }
+        }
+    }
+
+    @Nested
+    class ToInternal {
+
+        @Nested
+        @TestInstance(PER_CLASS)
+        class IncludeInternalBooleanValueTest extends InternalTextValueTest {
```
Comment (mww-sopra-steria) [inline ts/ts-bas/src/test/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionBeskrivningRiskfaktorerTest.java:213]: Rename class :-)

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nice! I added a few comments based on our discussions. (Maybe some have already been fixed before I submitted this review :-) )

File: ts/ts-bas/src/test/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionBeskrivningRiskfaktorerTest.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/test/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionBeskrivningRiskfaktorerTest.java:213]: Done!

File: support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/value/InternalValuePairTest.java [hunk 2]
```diff
@@ -29,11 +29,11 @@
 import se.inera.intyg.common.support.facade.model.Certificate;
 import se.inera.intyg.common.support.facade.model.CertificateDataElement;
 
-public abstract class InternalCodeListValueTest<T, S> {
+public abstract class InternalValuePairTest<T, S> {
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/value/InternalValuePairTest.java:32]: Done!

File: support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigRadioBooleanMultipleCodeTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    protected abstract List<RadioMultipleCode> getExpectedRadioMultipleCodes();
+
+    protected abstract Layout getExpectedLayout();
+
+    @Test
+    void shouldIncludeExpectedRadioMultipleCode() {
+        final var config = (CertificateDataConfigRadioMultipleCode) getElement().getConfig();
+        for (RadioMultipleCode radioMultipleCode : getExpectedRadioMultipleCodes()) {
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigRadioBooleanMultipleCodeTest.java:46]: Done!

File: support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigCheckboxMultipleCodeTest.java [hunk 4]
```diff
@@ -40,22 +41,11 @@ protected CertificateDataConfigTypes getType() {
...
-    }
-
-    @Test
-    void shouldIncludeConfigLabel() {
-        final var question = getElement();
-        final var config = (CertificateDataConfigCheckboxMultipleCode) question.getConfig();
-        for (int i = 0; i < config.getList().size(); i++) {
-            final var checkboxMultipleCode = config.getList().get(i);
-            assertEquals(checkboxMultipleCode.getLabel(), getListOfCodes().get(i).getLabel());
+        for (CheckboxMultipleCode checkboxMultipleCode : getExpectedListOfCodes()) {
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigCheckboxMultipleCodeTest.java:47]: Done!

File: support/src/main/java/se/inera/intyg/common/support/facade/model/config/Layout.java [hunk 5]
```diff
@@ -20,5 +20,5 @@
 package se.inera.intyg.common.support.facade.model.config;
 
 public enum Layout {
-    ROWS, COLUMNS, INLINE
+    ROWS, COLUMNS, COLUMN, INLINE
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply support/src/main/java/se/inera/intyg/common/support/facade/model/config/Layout.java:23]: Yes!

**Review by mww-sopra-steria — APPROVED** [review]
Great job! Approved!

**Pull request merged**